tests/prometheus/basic.py

```diff
@@ -0,0 +1,184 @@
+import asyncio
+from typing import Any, Optional, Type
+from unittest.mock import ANY, Mock, call
+
+import pytest
+from prometheus_client import CollectorRegistry
+
+from faststream.broker.core.usecase import BrokerUsecase
+from faststream.broker.message import AckStatus, StreamMessage
+from faststream.exceptions import RejectMessage
+from faststream.prometheus.middleware import (
+    PROCESSING_STATUS_BY_ACK_STATUS,
+    PROCESSING_STATUS_BY_HANDLER_EXCEPTION_MAP,
+    BasePrometheusMiddleware,
+)
+from faststream.prometheus.types import ProcessingStatus
+from tests.brokers.base.basic import BaseTestcaseConfig
+
+
+@pytest.mark.asyncio
+class LocalPrometheusTestcase(BaseTestcaseConfig):
+    broker_class: Type[BrokerUsecase]
+    middleware_class: Type[BasePrometheusMiddleware]
+    message_class: Type[StreamMessage[Any]]
+
+    @staticmethod
+    def consume_destination_name(queue: str) -> str:
+        return queue
+
+    @property
+    def settings_provider_factory(self):
+        return self.middleware_class(
+            registry=CollectorRegistry()
+        )._settings_provider_factory
+
+    @pytest.mark.parametrize(
+        (
+            "status",
+            "exception_class",
+        ),
+        [
+            (AckStatus.acked, RejectMessage),
+            (AckStatus.acked, Exception),
+            *[(status, None) for status in AckStatus],
+        ],
+    )
+    async def test_metrics(
```

Comment:
In my opinion, the mock-based approach to metric tests does not check the metrics themselves in any way and does not show reference values. An instance of the metric has a `collect()` method that outputs the final data. The approach to metric tests can be viewed here: https://github.com/draincoder/asgi-monitor/blob/develop/tests/unit/metrics/test_metrics.py

Reply:
done

Reply:
tests: https://github.com/airtai/faststream/pull/1791/files#diff-3965660aa83290de1b0b05e5690a8f38cbe25f7aa09337dc78102d88d32eab66